# Running MesKit on acral melanoma samples from Brazil

This repository is a sub-analysis and is included with other analyses in the [Acral Melanoma PDX models from Latin America parent](https://github.com/team113sanger/Acral_Melanoma_PDX_models_from_Latin_America) repository.

## Overview

This repository contains the code that was used to compare the patient and PDX-derived samples in terms of their somatic variants and copy number alteration (CNA) events.

## Repository structure

```bash
.
├── data
│   ├── copy_number
│   │   ├── combined_Human+PDX_with_1st+AltModels_segments_for_meskit.tsv
│   │   ├── ploidy_table_jan2025_data.tsv
│   │   └── segments
│   └── variants
│       ├── 6633_2729_3248-filtered_mutations_matched_allTum_keep_adjusted_for_meskit.maf
│       ├── 6633_2729_3248-filtered_mutations_matched_allTum_keep_clinical_for_meskit.tsv
│       └── 6633_2729_3248-filtered_mutations_matched_allTum_keep.maf
├── metadata
│   └── 6633_METADATA_table.tsv
├── README.md
├── renv.lock
├── results
│   ├── copy_number
│   │   ├── PD53330
│   │   ├── PD53332
│   │   ├── PD53333
│   │   ├── PD53337
│   │   ├── PD53343
│   │   ├── PD53347
│   │   ├── PD53349
│   │   ├── PD53350
│   │   ├── PD53352
│   │   ├── PD53355
│   │   ├── PD53357
│   │   ├── PD53359
│   │   └── PD53364
│   └── variants
│       ├── patient_vs_x1.csv
│       ├── PD53330
│       ├── PD53332
│       ├── PD53333
│       ├── PD53337
│       ├── PD53343_(A)
│       ├── PD53343_(B)
│       ├── PD53347
│       ├── PD53349_(A)
│       ├── PD53349_(B)
│       ├── PD53350_(A)
│       ├── PD53350_(B)
│       ├── PD53352_(A)
│       ├── PD53352_(B)
│       ├── PD53357
│       ├── PD53359
│       ├── PD53364
│       └── README.md
└── src
    ├── copy_number
    │   ├── 01-prepare_sequenza_for_meskit.R
    │   └── 02-make_cna_heatmaps.R
    └── variants
        ├── 01-adjust_maf_for_meskit.sh
        ├── 02-make_trees.R
        └── 03-compare_patient_and_x1.R
```

## Data

The `data/` directory contains most of the input files that are required to run the scripts in this repository, including somatic variant calls generated by CaVEMan/cgpPindel and segments output by Sequenza.

### Variant calls in `data/variants/`

- `6633_2729_3248-filtered_mutations_matched_allTum_keep.maf`: this is the MAF file with the variant calls that were used as input in the MesKit analysis; this file can also be obtained from [FigShare](https://figshare.com/account/items/29114711/edit).

- `6633_2729_3248-filtered_mutations_matched_allTum_keep_adjusted_for_meskit.maf` and `6633_2729_3248-filtered_mutations_matched_allTum_keep_clinical_for_meskit.tsv`: these files were generated by the script `01-adjust_maf_for_meskit.sh` using the original MAF `6633_2729_3248-filtered_mutations_matched_allTum_keep.maf`.

### Copy number data in `data/copy_number/`

- `segments/`: this subdirectory contains Sequenza segments, which can also be downloaded from [FigShare](https://figshare.com/articles/dataset/Sequenza_output_results_data_from_Brazilian_Acral_melanoma_samples_jan2025_/29088173?file=54607436). 

- `ploidy_table_jan2025_data.tsv`: this file contains the ploidy values for each sample.

- `combined_Human+PDX_with_1st+AltModels_segments_for_meskit.tsv`: this file was generated by the script `01-prepare_sequenza_for_meskit.R` using the segments under `data/copy_number/segments/`

### COSMIC data

The Cancer Gene Census (CGC) data from [COSMIC](https://cancer.sanger.ac.uk/cosmic) is licensed and therefore not included in this repository, but it can be retrieved from the link. Specifically, the file `cancer_gene_census.v97.csv` is expected by the script `02-make_trees.R` (more details on run to run the scripts present in this repository are explained in the `Analysis` section below).

## Required software and environment

The following software is required to run the analysis:

- R (version 4.2.2) - details on how to install it can be found [here](https://cran.r-project.org).
- R packages:
    - `MesKit`
    - `here`
    - `ggpubr`
    - `dplyr`
    - `tidyr`
    - `stringr`
    - `logger`
    - `readr`
    - `fs`

The required packages can be installed using:
```R
install.packages(c("MesKit", "here", "ggpubr", "dplyr", "tidyr", "stringr", "logger", "readr", "fs"))
```

If you are interested in reproducing the R environment for the code used in R 4.2.2, an `renv.lock` file is available in this repository. To rebuild the environment, run the following inside R:
```R
# Install renv if needed
if (!require("renv")) install.packages("renv")

# Restore the project environment
renv::restore()
```

## Analysis

### Phylogenetic trees and heatmaps from somatic variant calls

```bash
# 1 - Prepare the input files for MesKit.
bash src/variants/01-adjust_maf_for_meskit.sh

# 2 - Make the phylogenetic trees and heatmaps.
Rscript src/variants/02-make_trees.R

# 3 - Compare patient and PDX-X1 samples.
Rscript src/variants/03-compare_patient_and_x1.R
```

The output files generated by `01-adjust_maf_for_meskit.sh` can be found in `data/variants/` as explained in the `Data` section above. The trees, heatmaps and tables generated by `02-make_trees.R` and the table created by `03-compare_patient_and_x1.R` can be found in `results/variants/[patient]/`.

### Copy number heatmaps

```bash
# 1 - Prepare the input files for MesKit.
Rscript src/copy_number/01-prepare_sequenza_for_meskit.R

# 2 - Make heatmaps.
Rscript src/copy_number/02-make_cna_heatmaps.R
```

The output file generated by `01-prepare_sequenza_for_meskit.R` can be found in `data/copy_number/` as explained in the `Data` section above. The heatmaps generated by `02-make_cna_heatmaps.R` can be found in `results/copy_number/[patient]/`.

## Contact

If you have any questions or comments about this repository, please contact:

- Jacqueline Boccacino (<jb62@sanger.ac.uk>)
- Martin del Castillo Velasco-Herrera (<mdc1@sanger.ac.uk>)